### PR TITLE
Feat/13 dedicated treeview

### PR DESCRIPTION
**Pull request by michaeljohnbennett**
Initial PR for the dedicated treeview. This phase ended up being quite a complex bit of rework and refactoring to do the following:

- Add in the new view container
- Upgrade the vscode libs to 1.66.x
- use the newer az/vscode libs.
- Removal of all the old Azure and codegen code that was sitting there.
- Fixing a lot of tests that broke as a result of upgrade and removal of code.
- Fixing mocks and showQuickPick code around the place. No idea why this broke. Dependency/import order hell 🔥 
- Added in new help list and fixed ordering and icons.

This PR I think should be merged down to develop as its quite big, then I will continue on this branch to finish the other views.

Right now network and help views are working in the new truffle view. context menus are still working on the main file explorer but not on the new mock/simple file explorer on the truffle view.

### CODE REVIEW

**Comment by michaeljohnbennett** [comment]
image

**Review by xhulz — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -37,6 +37,7 @@
   "main": "./out/src/extension.js",
   "activationEvents": [
     "onView:truffle-vscode.truffle",
+    "onView:truffle-vscode.explorer-view",
```
Comment (kevinbluer) [inline package.json:40]: As discussed, for the explorer view if we could just show either just contents of the `contracts` directory (or ideally any directories that contain contracts if the project is using a monorepo, although appreciate that's way less trivial) that would awesome. Also for the right-click context menus, which look like they have to be manually wired up, the current commands that we display for `.sol`'s, e.g. `Build Contracts`, `Deploy Contracts`, etc

File: package.json [hunk 2]
```diff
@@ -443,10 +503,28 @@
...
+					"name": "Explorer"
+				},
         {
           "id": "truffle-vscode.truffle",
-          "name": "Truffle - Networks"
+          "name": "Networks",
+          "contextualTitle": "List of all known Networks"
+        },
+        {
+          "id": "truffle-vscode.views.deployments",
```
Comment (kevinbluer) [inline package.json:517]: Excluding this (and Libraries) for this upcoming release would be cool by me 👌

File: package.json [hunk 3]
```diff
@@ -443,10 +503,28 @@
...
+          "name": "Deployments"
+        },
+        {
+          "id": "truffle-vscode.views.libraries",
```
Comment (kevinbluer) [inline package.json:521]: As above, probably makes sense to exclude this for the this release

**Review by kevinbluer — CHANGES_REQUESTED** [review]
Awesome stuff @michaeljohnbennett! Few suggested tweaks (most of which we'd already discussed I think) 🙌

File: package.json (hunk 1 above)
Comment (michaeljohnbennett) [reply package.json:40]: contract dir showing, still need to work on the build deploy button integration/context menus

File: package.json (hunk 2 above)
Comment (michaeljohnbennett) [reply package.json:460]: got deployments working, have dropped libs

**Comment by michaeljohnbennett** [comment]
@xhulz do you think you will be able to give me a review of this code sometime over the next day please?

**Comment by michaeljohnbennett** [comment]
ok I will investigate the refresh on that, I need to see how to get a listener on the files/dir

**Comment by michaeljohnbennett** [comment]
> @michaeljohnbennett looks great!
> 
> Maybe we can put a simple refresh in the end of build / deploy command. Today, i think the networks treeview is working that way, calling this command refreshTree()

I added a refresh button now on the explorer view and a new command, I guess it could be added into the post deploy command task but would it be updated (the File) in time for the treeview to see the change? It depends on the network time to deploy etc and the task completing. Not sure if its async/sync.

But the command will do for a manual refresh and eventually to be triggered via the deploy task.

File: src/Models/TreeItems/HelpTreeItem.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    // default sorting is based on the label which is being displayed to user.
+    // use id to control the order being dispalyed
+    return item1!.id!.localeCompare(item2!.id!);
+  }
+
+  private get readGettingStartedGuidTreeItem(): AzExtTreeItem {
+    return new OpenUrlTreeItem(
+      this,
+      "0",
+      "Getting Started Guide",
```
Comment (xhulz) [inline src/Models/TreeItems/HelpTreeItem.ts:38]: In the future, we can put all these texts into a json file to simplify. What do you think?

**Review by xhulz — APPROVED** [review]
I've tested again and all is working smooth! Looks great to me!
Approved :)

File: src/Models/TreeItems/HelpTreeItem.ts (hunk 4 above)
Comment (michaeljohnbennett) [reply src/Models/TreeItems/HelpTreeItem.ts:38]: yes, I think once this is all happy we can move to a single spot for constants used in UI, internal etc.

**Pull request merged**